### PR TITLE
Feature/add maven local repository

### PR DESCRIPTION
**Pull request by jbern0rd**
Add mavenLocal() to find dependencies in local repository
Set target compatibility
Simplify library jar generation and publishing

### CODE REVIEW

**Comment by mcornaton** [comment]
Great! Can we remove this line and the following (region "Dependencies required by `iexec-common` when using local jar")?
https://github.com/iExecBlockchainComputing/iexec-blockchain-adapter-api/blob/59b9e64860286d946f611f785e8be369b9d958dc/build.gradle#L57

File: build.gradle [hunk 1]
```diff
@@ -90,19 +92,15 @@ task itest(type: Test) {
...
-//Create "libraryJar"
-task libraryJar(type: Jar, dependsOn: classes) {
-	archiveClassifier = 'library'
-	from sourceSets.main.allSource
-}
-//Create "bootJar" already done (so "jar" task can be disabled to avoid publish conflicts)
 jar {
-	enabled = false
+	enabled = true
+	archiveClassifier.set('library')
```
Comment (jeremyjams) [inline build.gradle:97]: How is it different from
`archiveClassifier = 'library'`
?

File: build.gradle [hunk 2]
```diff
@@ -13,6 +13,7 @@ ext {
 
 group = 'com.iexec.blockchain'
 sourceCompatibility = '11'
+targetCompatibility = '11'
```
Comment (jeremyjams) [inline build.gradle:16]: Just a question, did you have any issues without it?

**Comment by zguesmi** [comment]
How is the jar pushed to local maven?

File: build.gradle (hunk 1 above)
Comment (jbern0rd) [reply build.gradle:97]: archiveClassifier is not a `String` property, it is a `Property<String>`, `archiveClassifier = 'library'` simply does not work.
Before, it was possible to write `classifier = 'library'` but this has been deprecated.

File: build.gradle (hunk 2 above)
Comment (jbern0rd) [reply build.gradle:16]: Nope, but I saw it was defined in all projects.
Should I revert it ?

Comment (jeremyjams) [reply build.gradle:16]: No it's great, I was just wondering :+1:

**Review by jeremyjams — APPROVED** [review]
:+1:

**Comment by jbern0rd** [comment]
> How is the jar pushed to `mavenLocal` repository?

For this project, the `maven-publish` was already configured. It is the plugin which allows to publish to the local repository.
Here, the goal is to be able to resolve dependencies like `iexec-common` **from** the local repository.
This is done by adding `mavenLocal()` in the `repositories` container

**Comment by jbern0rd** [comment]
> Great! Can we remove this line and the following (region "Dependencies required by `iexec-common` when using local jar")?
> 
> https://github.com/iExecBlockchainComputing/iexec-blockchain-adapter-api/blob/59b9e64860286d946f611f785e8be369b9d958dc/build.gradle#L57

Done

**Review by mcornaton — APPROVED** [review]
Really nice, thanks!

**Review by zguesmi — APPROVED** [review]

**Pull request merged**